### PR TITLE
Link bug

### PR DESCRIPTION
**Pull request by manas-metron**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py [hunk 1]
```diff
@@ -27,8 +26,10 @@
 LINQ_LINK_BASE = demisto.params().get("linq_link_base", "https://us.devo.com/welcome")
 FETCH_INCIDENTS_FILTER = demisto.params().get("fetch_incidents_filters", None)
 FETCH_INCIDENTS_LIMIT = demisto.params().get("fetch_incidents_limit") or 50
+FETCH_INCIDENTS_LOOKBACK_SECONDS = demisto.params().get("fetch_incidents_lookback_seconds") or 3600
 # Deprecated: this parameter is never used
 FETCH_INCIDENTS_DEDUPE = demisto.params().get("fetch_incidents_deduplication", None)
+FETCH_INCIDENTS_WINDOW = demisto.params().get("fetch_incidents_window")
```
Comment (kapil-metron) [inline Packs/Devo/Integrations/Devo_v2/Devo_v2.py:32]: Is this mandatory param? If not, do we need a default value for this?

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py [hunk 2]
```diff
@@ -467,36 +452,63 @@ def fetch_incidents():
...
+
     # de duplicate events between two consecutive fetches
     if "last_fetch_events" in last_run:
-        last_events = last_run.get("last_fetch_events", [])
+        # Retrieve the alert Ids that is stored in the dictionary
+        last_events_ids = []
+        for record in last_run.get("last_fetch_events"):
+            for key in record:
+                last_events_ids.append(key)
+        demisto.debug(last_events_ids)
```
Comment (kapil-metron) [inline Packs/Devo/Integrations/Devo_v2/Devo_v2.py]: Please write some message for the Debug messages. Example:
demisto.debug(f"List of event ids fetched in last poll: {last_events_ids}")

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py [hunk 3]
```diff
@@ -467,36 +452,63 @@ def fetch_incidents():
...
+            if(timestamp > from_time):
+                from_previous_poll.append(record)
+
+    new_last_run["last_fetch_events"] = from_previous_poll
+
     demisto.setLastRun(new_last_run)
 
+    demisto.debug(f"ts: {time.time()} | func: fetch_incidents | msg: last_run_set = {new_last_run}")
+    demisto.debug(f"ts: {time.time()} | func: fetch_incidents | msg: number of incidents generated {len(incidents)}")
+
```
Comment (kapil-metron) [inline Packs/Devo/Integrations/Devo_v2/Devo_v2.py:511]: Can we restructure this function as follows:

- create_alert_query()
- calculate_fetch_time_range()
- fetch_alerts_for_given_timerange_and_filter()
- remove_duplicates_events_from_last_polled()
- dump_old_events_from_persistent_starage()

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py [hunk 4]
```diff
@@ -801,7 +813,21 @@ def write_to_table_command():
         "ReadableContentsFormat": formats["markdown"],
         "EntryContext": {"Devo.QueryLink": createContext(querylink)},
     }
-    md = tableToMarkdown("Entries to load into Devo", records)
+    headers:list = []
```
Comment (kapil-metron) [inline Packs/Devo/Integrations/Devo_v2/Devo_v2.py]: Can we have this as a separate function? and call it here, please also add comments about what it does.

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py [hunk 5]
```diff
@@ -903,7 +930,7 @@ def main():
             total = 0
             demisto.results(multi_table_query_command(OFFSET, items_per_page))
             total = total + COUNT_MULTI_TABLE
-            while COUNT_MULTI_TABLE == items_per_page * 2:
+            while items_per_page * 2 == COUNT_MULTI_TABLE:
```
Comment (kapil-metron) [inline Packs/Devo/Integrations/Devo_v2/Devo_v2.py:946]: What is this change for?

**Review by kapil-metron — CHANGES_REQUESTED** [review]

File: Packs/Devo/Integrations/Devo_v2/Devo_v2.py (hunk 1 above)
Comment (manas-metron) [reply Packs/Devo/Integrations/Devo_v2/Devo_v2.py:32]: It is not a required parameter. We only utilise it when it is set.

**Pull request merged**